histomicstk/preprocessing/positive_pixel_count.py

```diff
@@ -0,0 +1,232 @@
+from __future__ import division
+
+from collections import namedtuple
+
+from dask import delayed
+import large_image
+import numpy as np
+
+from .color_conversion import rgb_to_hsi
+
+
+# This can be an enum in Python >= 3.4
+class Labels(object):
+    """Labels for the output image of the positive pixel count routines."""
+    NEGATIVE = 0
+    WEAK = 1
+    PLAIN = 2
+    STRONG = 3
+
+
+class PPCParameters(
+        namedtuple('PPCParameters', [
+            'hue_value',
+            'hue_width',
+            'saturation_minimum',
+            'intensity_upper_limit',
+            'intensity_weak_threshold',
+            'intensity_strong_threshold',
+            'intensity_lower_limit',
+        ]),
+):
+    """PPCParameters(hue_value, hue_width, saturation_minimum,
+    intensity_upper_limit, intensity_weak_threshold,
+    intensity_strong_threshold, intensity_lower_limit)
+
+    Attributes
+    ----------
+    hue_value:
+      Center of the hue range in HSI space for the positive color, in
+      the range [0, 1]
+    hue_width:
+      Width of the hue range in HSI space
+    saturation_minimum:
+      Minimum saturation of positive pixels in HSI space, in the range
+      [0, 1]
+    intensity_upper_limit:
+      Intensity threshold in HSI space above which a pixel is
+      considered negative, in the range [0, 1]
+    intensity_weak_threshold:
+      Intensity threshold in HSI space that separates weak-positive
+      pixels (above) from plain positive pixels (below)
+    intensity_strong_threshold:
+      Intensity threshold in HSI space that separates plain positive
+      pixels (above) from strong positive pixels (below)
+    intensity_lower_limit:
+      Intensity threshold in HSI space below which a pixel is
+      considered negative
+
+    """
+
+
+PPCOutputTotals = namedtuple('PPCOutputTotals', [
+    'NumberWeakPositive',
+    'NumberPositive',
+    'NumberStrongPositive',
+    'IntensitySumWeakPositive',
+    'IntensitySumPositive',
+    'IntensitySumStrongPositive',
+])
+
+PPCOutput = namedtuple('PPCOutput', PPCOutputTotals._fields + (
+    'IntensityAverage',
+    'RatioStrongToTotal',
+    'IntensityAverageWeakAndPositive',
+))
```

Comment:
I feel we can combine PPCOutputTotals and PPCOutput into one. What's the reason to keep them separate?